src/dos/drive_virtual.cpp

```diff
@@ -79,7 +79,20 @@ VFILE_Block::~VFILE_Block()
 	}
 }
 
-char *VFILE_Generate_8x3(const char *name, const unsigned int onpos)
+// this gets replaced with std::find_if later
+template <typename Predicate>
+vfile_block_t find_file_by_predicate(vfile_block_t first_file, Predicate predicate_)
+{
+	auto cur_file = first_file;
+	while (cur_file) {
```

Comment:
This `first_file` here is visually ambiguous (and shadows) the global linked list:

![2022-12-14_09-22](https://user-images.githubusercontent.com/1557255/207664082-fcb95736-3416-47b4-8082-b4250cdc0c0a.png)

Reply:
This function scans a vfile-linked-list based on a template condition; suggest using more generic names indicating this (specifically clarifying these are vfiles, not normal files):
 - Maybe something like: `find_vfile_by_predicate()`, and
 - `first_file` could be `first_vfile` or `head_vfile`
 - `cur_file` could then just be `vfile`.

Reply:
good idea